hotp/hotp.go

```diff
@@ -57,27 +57,23 @@ type ValidateOpts struct {
 	Algorithm otp.Algorithm
 }
 
-// ValidateCustom validates an HOTP with customizable options. Most users should
-// use Validate().
-func ValidateCustom(passcode string, counter uint64, secret string, opts ValidateOpts) (bool, error) {
-	passcode = strings.TrimSpace(passcode)
-
-	switch opts.Digits {
-	case otp.DigitsSix:
-		if len(passcode) != 6 {
-			return false, otp.ErrValidateInputInvalidLength6
-		}
-	case otp.DigitsEight:
-		if len(passcode) != 8 {
-			return false, otp.ErrValidateInputInvalidLength8
-		}
-	default:
-		panic("unsupported Digits value.")
-	}
+// GenerateCode creates a HOTP passcode given a counter and secret.
+// This is a shortcut for GenerateCodeCustom, with parameters that
+// are compataible with Google-Authenticator.
+func GenerateCode(secret string, counter uint64) string {
```

Comment:
Not returning error here is the only thing i'm not sure about.

For the simple `Validate` its a little easier to reason about only returning a `bool` and no `error` -- you only care if the  token was valid from a flow perspective, if it was malformed, or whatever, you want to show the user the invalid token flow -- for Generate, if the secret is malformed, returning an empty string.... is pretty rough.  So i'm thinking it's return signature should be `(string, error)`